web_poet/testing/pytest.py

```diff
@@ -0,0 +1,132 @@
+import operator
+from pathlib import Path
+from typing import Any, Iterable, List, Optional, Set, Union
+
+import pytest
+
+from web_poet.testing.utils import OUTPUT_FILE_NAME, Fixture
+
+# https://github.com/pytest-dev/pytest/discussions/10261
+_version_tuple = getattr(pytest, "version_tuple", None)
+_new_pytest = _version_tuple and _version_tuple[0] >= 7
+
+
+class WebPoetFile(pytest.File):
+    """Represents a directory containing test subdirectories for one Page Object."""
+
+    @staticmethod
+    def sorted(items: List["WebPoetItem"]) -> List["WebPoetItem"]:
+        """Sort the test list by the test name."""
+        return sorted(items, key=operator.attrgetter("name"))
+
+    def collect(self) -> Iterable[Union[pytest.Item, pytest.Collector]]:  # noqa: D102
+        result: List[WebPoetItem] = []
+        path = self.path if _new_pytest else Path(self.fspath)
+        for entry in path.iterdir():
+            if entry.is_dir():
+                item: WebPoetItem = _get_item(
+                    self,
+                    name=entry.name,
+                    path=entry,
+                )
+                if item.fixture.is_valid():
+                    result.append(item)
+        return self.sorted(result)
+
+
+class WebPoetItem(pytest.Item):
+    """Represents a directory containing one test."""
+
+    if _new_pytest:
+
+        def __init__(
+            self,
+            name,
+            parent=None,
+            config: Optional["pytest.Config"] = None,
+            session: Optional[pytest.Session] = None,
+            nodeid: Optional[str] = None,
+            **kw,
+        ) -> None:
+            super().__init__(name, parent, config, session, nodeid, **kw)
+            self.fixture = Fixture(self.path)
+
+    else:
+
+        def __init__(  # type: ignore[misc]
```

Comment:
Do you know why is it marked as not covered, while there is an env with older pytest?

Reply:
Oh, because it runs in "check", not in "test" and so coverage for it is not uploaded. Not sure how to rewrite this in the best way, as the "test" steps run `tox -e py`.